Dev

bbkiwi opened the pull request:
Hi @cnlohr and @con-f-use 

       I'll create this now. @con-f-use has given me a few pointers about working in dev. 
       I realize all this is a mix of many changes and it would have been better to make pull requests for
       separate issues.  

      The first commit has a long description which outlines what I have been doing.  The
       other commits tweak and fix up what I tried to do. 

       But I thought it finally time to start this discussion. At this point I've can drive the 18 LEDs
       of my banjo uke in a variety of patterns using your routines with mods I've made. Here is 
       a small clip where it is picking up the sounds of nearby instruments.
                   https://youtu.be/mEXjZC2-_sA?t=1m33s

      What is not yet working is connecting to Stations. It makes an AP fine, but won't remember
          station info I put in the web gui. At this point I must first run a arduino sketch which sets
          up a   station.  Then when I flash it remembers the station info put in by that sketch.

      I'm happy to try and make things easy, so I could make a branch with the new sort routine
      etc. If that might help. 

      Cheers, Bill

cnlohr commented:
Woah there's a lot of stuff going on here. @con-f-use I am unfamiliar with how this would work on a project without a pre-existing "dev" branch.  Can I safely hit merge then test in post?

Also, @bbkiwi one major thing I wanted to note is now esp82xx supports binary websockets, you can transfer full-on binary data. (just fyi)

con-f-use commented:
I'd test it via `git clone --recursive https://github.com/bbkiwi/colorchord.git -b dev `. You could also create a `dev` branch and then merge it into that or you can merge it into `master` and revert, or undo the commit, if it's not to your liking. Last think is frowned upon because it fucks up repositories cloned after the merge. [TMTOWTDI™](https://en.wikipedia.org/wiki/There%27s_more_than_one_way_to_do_it)

cnlohr commented:
Got it.  This may also be the time to add support for RGBY LEDs.

cnlohr commented:
Also @bbkiwi I think I know why you're having issues with connecting to another station and will work on that.

bbkiwi commented:
Hi @con-f-use 
     This all started with issue #37 where I describe problems I was having with colorchord. I only started to look at esp82xx when it seemed some of the problems were coming from it.  At the start of issue #37 I explain my (lack of) background to all this, so I apologize if I don't quite know the best way to communicate. Any suggestions appreciated. 

Cheers, Bill

bbkiwi commented:
Hi @cnlohr 

     Solving the connection problem would be great. It is happening when I test the basic project in esp82xx/master.

opus506 commented:
Anyone try merging everything yet?
I'm going to try and merge to a dev branch tonight if not.

cnlohr commented:
gaah I didn't get to testing this tonight.  Please try and check.

cnlohr commented:
I did get to checking the new esp82xx... but... not cc.

Also, we really need to make sure cc and esp82xx are updated in pair.

bbkiwi commented:
if I try cc/dev with all my changes using most recent esp82xx/dev overflows by 1292 bytes. 

... /home/bill/82XX-projects/colorchord/embedded8266/esp82xx/toolchain/esp_nonos_sdk/lib  -o image.elf
...
region `iram1_0_seg' overflowed by 1292 bytes

cnlohr commented:
@con-f-use do you mind making sure my merging here won't wreck things up? 

Also @bbkiwi AFASF#$#(%G#%(G#(#(D@)C(ASD(V I thought we had enough memory recovered in the custom SDK I had pieced together.  UGH.

bbkiwi commented:
I wasn't sure if AFASF ... was an acronym or just your left hand going mad :-).
I watched a bit of your youtube hack fest on Sat and noticed that submodules git you some grief too :-)

Looking issue #35 where make all embedd8286 overflowed by 288 bytes. 
 I just tried to compile using be144 commit from colorchord master with submodules esp82xx newest (51c38) commit of cnlohr/esp82xx master and newest commit (7d66f) for esp_nonos_sdk  and still get `iram1_0_seg' overflowed by 288 bytes.  

Note I only have tested my changes to embeddedcommon for embedded8266 (with 18 LEDs) and embeddedlinux. I have not way to try it for the other two embedded devices.

cnlohr commented:
@bbkiwi are you up for pursuing this again?  I don't know any way around it other than manually implementing most of these changes.  Is that something I should do?

Will you always have a separate trunk without that?

bbkiwi commented:
I'm away from home for a few more days and then will have access to my computer. Might be a good idea to skype to discuss. Both branches on my fork,  adddevwork and newfeatures which use slightly modified esp82xx submodule which use your custom sdk have no memory problems.

cnlohr commented:
Just wondering... what came of all of this?  I thought it was mostly integrated another way.

Fireflaker commented:
[colorchord.zip](https://github.com/cnlohr/colorchord/files/2259192/colorchord.zip)
Colorchord.exe for ones who suffer to see it working.

cnlohr commented:
Do you mind elaborating more?  I don't really understand what happened with this still.

con-f-use commented:
From what I can see most of the stuff that changed something in esp82xx has been merged into esp82xx, right?

bbkiwi commented:
Hi @cnlohr  and @con-f-use, I lost track a bit about all this and had not done much for a long time, but just recently started again. Also my repo was screwed up so a clone of it couldn't bring in the submodule using my fork of exp82xx correctly. However I think I've fixed that all. 

In my cc fork https://github.com/bbkiwi/colorchord/commit/0717767ac8a2e8163e61a942cc0fc43013ebdbef I'm using newfeatures branch which uses esp82xx submodule from my fork of esp82xx at the top of branch tmp. I've compared  https://github.com/bbkiwi/esp82xx/commit/75b0773fe0d70be92417cab68b060727019ecd96  with https://github.com/cnlohr/esp82xx/commit/b7fc6f637ae9c3302195f54572e1cc5d007124f1and there are some differences but the only one I notice that might affect the interaction of ADC and wifi is that I call StartHPATimer in GoAP in commonservices.c. Other changes I made were to add an option to assign fixed IPs. Other changes are more cosmetic. 

At this point using my fork on branch newfeatures  is working well and I will probably continue using it. 
I've mainly made changes to embedded8266 as my interest is using cc on the ESP8266. 

It is stable tested on 5 Nodemcu and Weimos too, I can connect to a specified IP, I've used it with LED rings from 16 to 60. 

I've added a 'glitch' filter to get rid of interferance of sockets with the ADC. 
I have tweaked the gui considerably adding slider controls for  many new configuration parameters giving finer control to how notes are generated (there were a few 'bugs' in finding notes in the sense that the code did not produce what comments seemed to suggest was happening) and displayed.  16 different sets of parameters can be saved in flash, the number of leds to be used (upto 256) is now a parameter, so re compiling is not necessary for this. 

I'm not sure it will be easy to make pull requests for these as the changes are not easily isolated. 
I'm almost done documenting all this which will explain in more detail.
